src/AppInstallerCLICore/Core.cpp

```diff
@@ -66,21 +67,34 @@ namespace AppInstaller::CLI
 
         init_apartment();
 
-#ifndef AICLI_DISABLE_TEST_HOOKS
+#ifndef AICLI_DISABLE_TEST_HOOKS
+        // We have to do this here so the auto minidump config initialization gets caught
+        Logging::OutputDebugStringLogger::Add();
+        Logging::Log().EnableChannel(Logging::Channel::All);
+        Logging::Log().SetLevel(Logging::Level::Verbose);
+
         if (Settings::User().Get<Settings::Setting::EnableSelfInitiatedMinidump>())
         {
             Debugging::EnableSelfInitiatedMinidump();
-        }
+        }
+
+        Logging::OutputDebugStringLogger::Remove();
 #endif
 
         Logging::UseGlobalTelemetryLoggerActivityIdOnly();
 
         Execution::Context context{ std::cout, std::cin };
-        auto previousThreadGlobals = context.SetForCurrentThread();
+        auto previousThreadGlobals = context.SetForCurrentThread();
+
+        // Set up debug string logging during initialization
+        Logging::OutputDebugStringLogger::Add();
```

Comment:
I don't see why we have to remove the ODS logger only to add it immediately after. Couldn't we just add it before the `#ifndef` and remove it once in line 97?

Reply:
The first one is added to the global logger, while the second one is added to the context logger.